File: C-Sharp/frmDepositTake.cs
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Windows.Forms;
using DpSdkEngLib;
using DPSDKOPSLib;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
 // ERROR: Not supported in C#: OptionDeclaration
namespace _4PosBackOffice.NET
{
	internal partial class frmDepositTake : System.Windows.Forms.Form
	{
		private ADODB.Recordset withEventsField_adoPrimaryRS;
		public ADODB.Recordset adoPrimaryRS {
			get { return withEventsField_adoPrimaryRS; }
			set {
				if (withEventsField_adoPrimaryRS != null) {
					withEventsField_adoPrimaryRS.MoveComplete -= adoPrimaryRS_MoveComplete;
					withEventsField_adoPrimaryRS.WillChangeRecord -= adoPrimaryRS_WillChangeRecord;
				}
				withEventsField_adoPrimaryRS = value;
				if (withEventsField_adoPrimaryRS != null) {
					withEventsField_adoPrimaryRS.MoveComplete += adoPrimaryRS_MoveComplete;
					withEventsField_adoPrimaryRS.WillChangeRecord += adoPrimaryRS_WillChangeRecord;
				}
			}
		}
		bool mbChangedByCode;
		int mvBookMark;
		bool mbEditFlag;
		bool mbAddNewFlag;
		bool mbDataChanged;

		string gFilter;
		string gFilterSQL;
		private StdFormat.StdDataFormat fmtBooleanData;

		private void loadLanguage()
		{

			//frmDepositTake = No Code   [Deposit Stock Take Adjustments]
			//'rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
			//If rsLang.RecordCount Then frmDepositTake.Caption = rsLang("LanguageLayoutLnk_Description"): frmDepositTake.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")

			//lblHeading = No Code       [Deposit Stock Take List]
			//rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
			//If rsLang.RecordCount Then lblHeading.Caption = rsLang("LanguageLayoutLnk_Description"): lblHeading.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")

			modRecordSet.rsLang.filter = "LanguageLayoutLnk_LanguageID=" + 1085;
			//Print|Checked
			if (modRecordSet.rsLang.RecordCount){cmdPrint.Text = modRecordSet.rsLang.Fields("LanguageLayoutLnk_Description").Value;cmdPrint.RightToLeft = modRecordSet.rsLang.Fields("LanguageLayoutLnk_RightTL").Value;}

			modRecordSet.rsLang.filter = "LanguageLayoutLnk_LanguageID=" + 1004;
			//Exit|Checked
			if (modRecordSet.rsLang.RecordCount){cmdClose.Text = modRecordSet.rsLang.Fields("LanguageLayoutLnk_Description").Value;cmdClose.RightToLeft = modRecordSet.rsLang.Fields("LanguageLayoutLnk_RightTL").Value;}

			modRecordSet.rsHelp.filter = "Help_Section=0 AND Help_Form='" + this.Name + "'";
			//UPGRADE_ISSUE: Form property frmDepositTake.ToolTip1 was not upgraded. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="CC4C7EC0-C903-48FC-ACCC-81861D12DA4A"'
			if (modRecordSet.rsHelp.RecordCount)
				this.ToolTip1 = modRecordSet.rsHelp.Fields("Help_ContextID").Value;

		}

		public void loadItem()
		{
			My.MyProject.Forms.frmStockTakeSnapshot.remoteSnapShot();
			System.Windows.Forms.Application.DoEvents();

			getNamespace();
			mbDataChanged = false;

			loadLanguage();
			ShowDialog();
		}


		private void cmdFilter_Click()
		{
			My.MyProject.Forms.frmFilter.loadFilter(ref gFilter);
			getNamespace();
		}

		private void getNamespace()
		{

			adoPrimaryRS = modRecordSet.getRS(ref "SELECT Deposit.Deposit_Name, StockTakeDeposit_DepositTypeID, StockTakeDeposit.StockTakeDeposit_Quantity, StockTakeDeposit_DepositTypeID, StockTakeDeposit.StockTakeDeposit_WarehouseID, StockTakeDeposit.StockTakeDeposit_DepositID FROM Deposit INNER JOIN StockTakeDeposit ON Deposit.DepositID = StockTakeDeposit.StockTakeDeposit_DepositID Where (((StockTakeDeposit.StockTakeDeposit_WarehouseID) = 2)) ORDER BY Deposit.Deposit_Name, StockTakeDeposit.StockTakeDeposit_DepositTypeID;");

			//Display the list of Titles in the DataCombo
			grdDataGrid.DataSource = adoPrimaryRS;
			grdDataGrid.Columns[0].HeaderText = "Stock Name";
			grdDataGrid.Columns[0].DefaultCellStyle.Alignment = MSDataGridLib.AlignmentConstants.dbgLeft;
			grdDataGrid.Columns[0].Frozen = true;

			grdDataGrid.Columns[1].HeaderText = "Type";
			grdDataGrid.Columns[1].Frozen = true;
			grdDataGrid.Columns[1].Width = sizeConvertors.twipsToPixels(900, true);
			grdDataGrid.Columns[1].DefaultCellStyle.Format = fmtBooleanData;


			grdDataGrid.Columns[2].HeaderText = "Quantity";
			grdDataGrid.Columns[2].DefaultCellStyle.Alignment = MSDataGridLib.AlignmentConstants.dbgRight;
			grdDataGrid.Columns[2].Width = sizeConvertors.twipsToPixels(900, true);
			//UPGRADE_WARNING: Couldn't resolve default property of object grdDataGrid.Columns().DataFormat.Type. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"'
			//grdDataGrid.Columns(2).DefaultCellStyle.FormatProvid = 1
			grdDataGrid.Columns[2].DefaultCellStyle.Format = "#,##0";
			grdDataGrid.Columns[2].Frozen = false;

			grdDataGrid.Columns[5].Visible = false;
			grdDataGrid.Columns[3].Visible = false;
			grdDataGrid.Columns[4].Visible = false;

			frmDepositTake_Resize(this, new System.EventArgs());
			mbDataChanged = false;

		}



		private void cmdPrint_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			modApplication.report_DepositTake();
		}

		private void frmDepositTake_Load(System.Object eventSender, System.EventArgs eventArgs)
		{
			fmtBooleanData = new StdFormat.StdDataFormat();
			fmtBooleanData.Type = StdFormat.FormatType.fmtBoolean;
			fmtBooleanData.TrueValue = "Crate";
			fmtBooleanData.FalseValue = "Bottle";
			fmtBooleanData.NullValue = "";

		}
		private void frmDepositTake_KeyPress(System.Object eventSender, System.Windows.Forms.KeyPressEventArgs eventArgs)
		{
			short KeyAscii = Strings.Asc(eventArgs.KeyChar);
			if (KeyAscii == 27) {
				KeyAscii = 0;
				cmdClose_Click(cmdClose, new System.EventArgs());
			}
			eventArgs.KeyChar = Strings.Chr(KeyAscii);
			if (KeyAscii == 0) {
				eventArgs.Handled = true;
			}
		}

//UPGRADE_WARNING: Event frmDepositTake.Resize may fire when form is initialized. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="88B12AE1-6DE0-48A0-86F1-60C0686C026A"'
		private void frmDepositTake_Resize(System.Object eventSender, System.EventArgs eventArgs)
		{
			 // ERROR: Not supported in C#: OnErrorStatement

			//This will resize the grid when the form is resized
			System.Windows.Forms.Application.DoEvents();
			grdDataGrid.Height = sizeConvertors.twipsToPixels(sizeConvertors.pixelToTwips(this.ClientRectangle.Height, false) - 30 - sizeConvertors.pixelToTwips(picButtons.Height, false), false);
			grdDataGrid.Columns[0].Width = sizeConvertors.twipsToPixels(sizeConvertors.pixelToTwips(grdDataGrid.Width, true) - 1800 - 580, true);

		}

		private void frmDepositTake_FormClosed(System.Object eventSender, System.Windows.Forms.FormClosedEventArgs eventArgs)
		{
			System.Windows.Forms.Cursor.Current = System.Windows.Forms.Cursors.Default;
		}

		private void adoPrimaryRS_MoveComplete(ADODB.EventReasonEnum adReason, ADODB.Error pError, ref ADODB.EventStatusEnum adStatus, ADODB.Recordset pRecordset)
		{
			//This will display the current record position for this recordset
		}

		private void adoPrimaryRS_WillChangeRecord(ADODB.EventReasonEnum adReason, int cRecords, ref ADODB.EventStatusEnum adStatus, ADODB.Recordset pRecordset)
		{
			int lQuantity = 0;
			if (adoPrimaryRS.Fields("StockTakeDeposit_Quantity").OriginalValue != adoPrimaryRS.Fields("StockTakeDeposit_Quantity").Value) {
				lQuantity = Convert.ToInt32(adoPrimaryRS.Fields("StockTakeDeposit_Quantity").Value) - Convert.ToInt32(adoPrimaryRS.Fields("StockTakeDeposit_Quantity").OriginalValue);
				modRecordSet.cnnDB.Execute("UPDATE WarehouseDepositItemLnk SET WarehouseDepositItemLnk.WarehouseDepositItemLnk_Quantity = [WarehouseDepositItemLnk]![WarehouseDepositItemLnk_Quantity]+(" + lQuantity + ") WHERE (((WarehouseDepositItemLnk.WarehouseDepositItemLnk_WarehouseID)=" + adoPrimaryRS.Fields("StockTakeDeposit_WarehouseID").Value + ") AND ((WarehouseDepositItemLnk.WarehouseDepositItemLnk_DepositID)=" + adoPrimaryRS.Fields("StockTakeDeposit_DepositID").Value + ") AND ((WarehouseDepositItemLnk.WarehouseDepositItemLnk_DepositTypeID)=" + adoPrimaryRS.Fields(1).Value + "));");

				modRecordSet.cnnDB.Execute("UPDATE Company INNER JOIN DayEndDepositItemLnk ON Company.Company_DayEndID = DayEndDepositItemLnk.DayEndDepositItemLnk_DayEndID SET DayEndDepositItemLnk.DayEndDepositItemLnk_QuantityShrink = [DayEndDepositItemLnk]![DayEndDepositItemLnk_QuantityShrink]+" + lQuantity + " WHERE (((DayEndDepositItemLnk.DayEndDeposittemLnk_DepositID)=" + adoPrimaryRS.Fields("StockTakeDeposit_DepositID").Value + ") AND ((DayEndDepositItemLnk.DayEndDeposittemLnk_DepositType)=" + adoPrimaryRS.Fields(1).Value + "));");

				doDiskFlush();

			}
		}
		private void doDiskFlush()
		{
			return;
			Scripting.FileSystemObject fso = new Scripting.FileSystemObject();
			int hkey = 0;
			int lRetVal = 0;
			string vValue = null;
			string lPath = null;
			ADODB.Recordset rs = default(ADODB.Recordset);
			int lID = 0;
			int lCompanyID = 0;
			string lString = null;
			string lKey = null;
			short lDepositType = 0;
			lID = adoPrimaryRS.Fields("StockTakeDeposit_DepositID").Value;
			lDepositType = adoPrimaryRS.Fields(1).Value;
			lRetVal = modUtilities.RegOpenKeyEx(modUtilities.HKEY_LOCAL_MACHINE, "Software\\4POS", 0, modUtilities.KEY_QUERY_VALUE, ref hkey);
			lRetVal = modUtilities.QueryValueEx(hkey, "master", ref vValue);
			modUtilities.RegCloseKey(hkey);
			if (string.IsNullOrEmpty(vValue)) {
				return;
			} else {
				lPath = vValue;
			}
			rs = modRecordSet.getRS(ref "SELECT Company.CompanyID, DayEndDepositItemLnk.DayEndDeposittemLnk_DepositID, DayEndDepositItemLnk.DayEndDepositItemLnk_DayEndID, DayEndDepositItemLnk.DayEndDepositItemLnk_QuantityShrink, DayEndDepositItemLnk.DayEndDeposittemLnk_DepositType FROM Company INNER JOIN DayEndDepositItemLnk ON Company.Company_DayEndID = DayEndDepositItemLnk.DayEndDepositItemLnk_DayEndID WHERE (((DayEndDepositItemLnk.DayEndDeposittemLnk_DepositID)=" + lID + ") AND ((DayEndDepositItemLnk.DayEndDeposittemLnk_DepositType)=" + lDepositType + "));");

			//If rs.RecordCount Then
			//Key = rs("CompanyID") & "_" & rs("DayEndDeposittemLnk_DepositID") & "_" & rs("DayEndDepositItemLnk_DayEndID") & "_" & CInt(lDepositType + 1)
			//lCompanyID = rs("CompanyID")
			//If fso.FileExists(lPath & lCompanyID & "\" & Key & ".stk") Then fso.DeleteFile lPath & lCompanyID & "\" & Key & ".stk"
			//If rs("DayEndDepositItemLnk_QuantityShrink") Then
			//Set lTextstream = fso.OpenTextFile(lPath & lCompanyID & "\" & Key & ".stk", ForWriting, True)
			//lTextstream.Write rs("DayEndDepositItemLnk_QuantityShrink")
			//lTextstream.Close
			//End If
			//End If

		}
		private void cmdCancel_Click()
		{
			 // ERROR: Not supported in C#: OnErrorStatement


			mbEditFlag = false;
			mbAddNewFlag = false;
			adoPrimaryRS.CancelUpdate();
			if (mvBookMark > 0) {
				adoPrimaryRS.Bookmark = mvBookMark;
			} else {
				adoPrimaryRS.MoveFirst();
			}
			mbDataChanged = false;

		}

//UPGRADE_NOTE: update was upgraded to update_Renamed. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="A9E4979A-37FA-4718-9994-97DD76ED70A7"'
		private void update_Renamed()
		{
			 // ERROR: Not supported in C#: OnErrorStatement


			adoPrimaryRS.UpdateBatch(ADODB.AffectEnum.adAffectAll);

			if (mbAddNewFlag) {
				adoPrimaryRS.MoveLast();
				//move to the new record
			}

			mbEditFlag = false;
			mbAddNewFlag = false;
			mbDataChanged = false;

			return;
			UpdateErr:
			Interaction.MsgBox(Err().Description);
		}

		private void cmdClose_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			update_Renamed();
			this.Close();
		}

		private void goFirst()
		{
			 // ERROR: Not supported in C#: OnErrorStatement


			adoPrimaryRS.MoveFirst();
			mbDataChanged = false;

			return;
			GoFirstError:

			Interaction.MsgBox(Err().Description);
		}

		private void goLast()
		{
			 // ERROR: Not supported in C#: OnErrorStatement


			adoPrimaryRS.MoveLast();
			mbDataChanged = false;

			return;
			GoLastError:

			Interaction.MsgBox(Err().Description);
		}

		//Private Sub grdDataGrid_CellValueChanged(ByVal eventSender As System.Object, ByVal eventArgs As AxMSDataGridLib.DDataGridEvents_CellValueChangedEvent) Handles grdDataGrid.CellValueChanged
		//    If grdDataGrid.Columns(ColIndex).DataFormat.Format = "#,##0.00" Then
		//       grdDataGrid.Columns(ColIndex).DataFormat = 0
		//    End If

		//End Sub
	}
}

File: VB/4PosBackOffice.NET/frmDepositTake.vb
Option Strict Off
Option Explicit On
Friend Class frmDepositTake
	Inherits System.Windows.Forms.Form
	Dim WithEvents adoPrimaryRS As ADODB.Recordset
	Dim mbChangedByCode As Boolean
    Dim mvBookMark As Integer
	Dim mbEditFlag As Boolean
	Dim mbAddNewFlag As Boolean
	Dim mbDataChanged As Boolean
	
	Dim gFilter As String
	Dim gFilterSQL As String
	Private fmtBooleanData As StdFormat.StdDataFormat
	
	Private Sub loadLanguage()
		
		'frmDepositTake = No Code   [Deposit Stock Take Adjustments]
		''rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
		'If rsLang.RecordCount Then frmDepositTake.Caption = rsLang("LanguageLayoutLnk_Description"): frmDepositTake.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
		
		'lblHeading = No Code       [Deposit Stock Take List]
		'rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 0000
		'If rsLang.RecordCount Then lblHeading.Caption = rsLang("LanguageLayoutLnk_Description"): lblHeading.RightToLeft = rsLang("LanguageLayoutLnk_RightTL")
		
		rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 1085 'Print|Checked
		If rsLang.RecordCount Then cmdPrint.Text = rsLang.Fields("LanguageLayoutLnk_Description").Value : cmdPrint.RightToLeft = rsLang.Fields("LanguageLayoutLnk_RightTL").Value
		
		rsLang.filter = "LanguageLayoutLnk_LanguageID=" & 1004 'Exit|Checked
		If rsLang.RecordCount Then cmdClose.Text = rsLang.Fields("LanguageLayoutLnk_Description").Value : cmdClose.RightToLeft = rsLang.Fields("LanguageLayoutLnk_RightTL").Value
		
		rsHelp.filter = "Help_Section=0 AND Help_Form='" & Me.Name & "'"
		'UPGRADE_ISSUE: Form property frmDepositTake.ToolTip1 was not upgraded. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="CC4C7EC0-C903-48FC-ACCC-81861D12DA4A"'
		If rsHelp.RecordCount Then Me.ToolTip1 = rsHelp.Fields("Help_ContextID").Value
		
	End Sub
	
	Public Sub loadItem()
		frmStockTakeSnapshot.remoteSnapShot()
		System.Windows.Forms.Application.DoEvents()
		
		getNamespace()
		mbDataChanged = False
		
		loadLanguage()
		ShowDialog()
	End Sub
	
	
	Private Sub cmdFilter_Click()
		frmFilter.loadFilter(gFilter)
		getNamespace()
	End Sub
	
	Private Sub getNamespace()
		
		adoPrimaryRS = getRS("SELECT Deposit.Deposit_Name, StockTakeDeposit_DepositTypeID, StockTakeDeposit.StockTakeDeposit_Quantity, StockTakeDeposit_DepositTypeID, StockTakeDeposit.StockTakeDeposit_WarehouseID, StockTakeDeposit.StockTakeDeposit_DepositID FROM Deposit INNER JOIN StockTakeDeposit ON Deposit.DepositID = StockTakeDeposit.StockTakeDeposit_DepositID Where (((StockTakeDeposit.StockTakeDeposit_WarehouseID) = 2)) ORDER BY Deposit.Deposit_Name, StockTakeDeposit.StockTakeDeposit_DepositTypeID;")
		
		'Display the list of Titles in the DataCombo
        grdDataGrid.DataSource = adoPrimaryRS
        grdDataGrid.Columns(0).HeaderText = "Stock Name"
        grdDataGrid.Columns(0).DefaultCellStyle.Alignment = MSDataGridLib.AlignmentConstants.dbgLeft
        grdDataGrid.Columns(0).Frozen = True
		
        grdDataGrid.Columns(1).HeaderText = "Type"
        grdDataGrid.Columns(1).Frozen = True
        grdDataGrid.Columns(1).Width = twipsToPixels(900, True)
        grdDataGrid.Columns(1).DefaultCellStyle.Format = fmtBooleanData
		
		
        grdDataGrid.Columns(2).HeaderText = "Quantity"
        grdDataGrid.Columns(2).DefaultCellStyle.Alignment = MSDataGridLib.AlignmentConstants.dbgRight
        grdDataGrid.Columns(2).Width = twipsToPixels(900, True)
		'UPGRADE_WARNING: Couldn't resolve default property of object grdDataGrid.Columns().DataFormat.Type. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="6A50421D-15FE-4896-8A1B-2EC21E9037B2"'
        'grdDataGrid.Columns(2).DefaultCellStyle.FormatProvid = 1
        grdDataGrid.Columns(2).DefaultCellStyle.Format = "#,##0"
        grdDataGrid.Columns(2).Frozen = False
		
		grdDataGrid.Columns(5).Visible = False
		grdDataGrid.Columns(3).Visible = False
		grdDataGrid.Columns(4).Visible = False
		
		frmDepositTake_Resize(Me, New System.EventArgs())
		mbDataChanged = False
		
	End Sub
	
	
	
	Private Sub cmdPrint_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdPrint.Click
		report_DepositTake()
	End Sub
	
	Private Sub frmDepositTake_Load(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles MyBase.Load
		fmtBooleanData = New StdFormat.StdDataFormat
		fmtBooleanData.Type = StdFormat.FormatType.fmtBoolean
		fmtBooleanData.TrueValue = "Crate"
		fmtBooleanData.FalseValue = "Bottle"
		fmtBooleanData.NullValue = ""
		
	End Sub
	Private Sub frmDepositTake_KeyPress(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.KeyPressEventArgs) Handles MyBase.KeyPress
		Dim KeyAscii As Short = Asc(eventArgs.KeyChar)
		If KeyAscii = 27 Then
			KeyAscii = 0
			cmdClose_Click(cmdClose, New System.EventArgs())
		End If
		eventArgs.KeyChar = Chr(KeyAscii)
		If KeyAscii = 0 Then
			eventArgs.Handled = True
		End If
	End Sub
	
	'UPGRADE_WARNING: Event frmDepositTake.Resize may fire when form is initialized. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="88B12AE1-6DE0-48A0-86F1-60C0686C026A"'
	Private Sub frmDepositTake_Resize(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles MyBase.Resize
		On Error Resume Next
		'This will resize the grid when the form is resized
		System.Windows.Forms.Application.DoEvents()
        grdDataGrid.Height = twipsToPixels(pixelToTwips(Me.ClientRectangle.Height, False) _
                                          - 30 - pixelToTwips(picButtons.Height, False), False)
        grdDataGrid.Columns(0).Width = twipsToPixels(pixelToTwips(grdDataGrid.Width, True) _
                                                    - 1800 - 580, True)
		
	End Sub
	
	Private Sub frmDepositTake_FormClosed(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.FormClosedEventArgs) Handles Me.FormClosed
		System.Windows.Forms.Cursor.Current = System.Windows.Forms.Cursors.Default
	End Sub
	
	Private Sub adoPrimaryRS_MoveComplete(ByVal adReason As ADODB.EventReasonEnum, ByVal pError As ADODB.Error, ByRef adStatus As ADODB.EventStatusEnum, ByVal pRecordset As ADODB.Recordset) Handles adoPrimaryRS.MoveComplete
		'This will display the current record position for this recordset
	End Sub
	
	Private Sub adoPrimaryRS_WillChangeRecord(ByVal adReason As ADODB.EventReasonEnum, ByVal cRecords As Integer, ByRef adStatus As ADODB.EventStatusEnum, ByVal pRecordset As ADODB.Recordset) Handles adoPrimaryRS.WillChangeRecord
        Dim lQuantity As Integer
		If adoPrimaryRS.Fields("StockTakeDeposit_Quantity").OriginalValue <> adoPrimaryRS.Fields("StockTakeDeposit_Quantity").Value Then
            lQuantity = CInt(adoPrimaryRS.Fields("StockTakeDeposit_Quantity").Value) - CInt(adoPrimaryRS.Fields("StockTakeDeposit_Quantity").OriginalValue)
			cnnDB.Execute("UPDATE WarehouseDepositItemLnk SET WarehouseDepositItemLnk.WarehouseDepositItemLnk_Quantity = [WarehouseDepositItemLnk]![WarehouseDepositItemLnk_Quantity]+(" & lQuantity & ") WHERE (((WarehouseDepositItemLnk.WarehouseDepositItemLnk_WarehouseID)=" & adoPrimaryRS.Fields("StockTakeDeposit_WarehouseID").Value & ") AND ((WarehouseDepositItemLnk.WarehouseDepositItemLnk_DepositID)=" & adoPrimaryRS.Fields("StockTakeDeposit_DepositID").Value & ") AND ((WarehouseDepositItemLnk.WarehouseDepositItemLnk_DepositTypeID)=" & adoPrimaryRS.Fields(1).Value & "));")
			
			cnnDB.Execute("UPDATE Company INNER JOIN DayEndDepositItemLnk ON Company.Company_DayEndID = DayEndDepositItemLnk.DayEndDepositItemLnk_DayEndID SET DayEndDepositItemLnk.DayEndDepositItemLnk_QuantityShrink = [DayEndDepositItemLnk]![DayEndDepositItemLnk_QuantityShrink]+" & lQuantity & " WHERE (((DayEndDepositItemLnk.DayEndDeposittemLnk_DepositID)=" & adoPrimaryRS.Fields("StockTakeDeposit_DepositID").Value & ") AND ((DayEndDepositItemLnk.DayEndDeposittemLnk_DepositType)=" & adoPrimaryRS.Fields(1).Value & "));")
			
			doDiskFlush()
			
		End If
	End Sub
	Private Sub doDiskFlush()
		Exit Sub
		Dim fso As New Scripting.FileSystemObject
        Dim hkey As Integer
		Dim lRetVal As Integer
		Dim vValue As String
		Dim lPath As String
		Dim rs As ADODB.Recordset
		Dim lID, lCompanyID As Integer
		Dim lString As String
		Dim lKey As String
		Dim lDepositType As Short
		lID = adoPrimaryRS.Fields("StockTakeDeposit_DepositID").Value
		lDepositType = adoPrimaryRS.Fields(1).Value
		lRetVal = RegOpenKeyEx(HKEY_LOCAL_MACHINE, "Software\4POS", 0, KEY_QUERY_VALUE, hkey)
		lRetVal = QueryValueEx(hkey, "master", vValue)
		RegCloseKey(hkey)
		If vValue = "" Then
			Exit Sub
		Else
			lPath = vValue
		End If
		rs = getRS("SELECT Company.CompanyID, DayEndDepositItemLnk.DayEndDeposittemLnk_DepositID, DayEndDepositItemLnk.DayEndDepositItemLnk_DayEndID, DayEndDepositItemLnk.DayEndDepositItemLnk_QuantityShrink, DayEndDepositItemLnk.DayEndDeposittemLnk_DepositType FROM Company INNER JOIN DayEndDepositItemLnk ON Company.Company_DayEndID = DayEndDepositItemLnk.DayEndDepositItemLnk_DayEndID WHERE (((DayEndDepositItemLnk.DayEndDeposittemLnk_DepositID)=" & lID & ") AND ((DayEndDepositItemLnk.DayEndDeposittemLnk_DepositType)=" & lDepositType & "));")
		
		'If rs.RecordCount Then
		'Key = rs("CompanyID") & "_" & rs("DayEndDeposittemLnk_DepositID") & "_" & rs("DayEndDepositItemLnk_DayEndID") & "_" & CInt(lDepositType + 1)
		'lCompanyID = rs("CompanyID")
		'If fso.FileExists(lPath & lCompanyID & "\" & Key & ".stk") Then fso.DeleteFile lPath & lCompanyID & "\" & Key & ".stk"
		'If rs("DayEndDepositItemLnk_QuantityShrink") Then
		'Set lTextstream = fso.OpenTextFile(lPath & lCompanyID & "\" & Key & ".stk", ForWriting, True)
		'lTextstream.Write rs("DayEndDepositItemLnk_QuantityShrink")
		'lTextstream.Close
		'End If
		'End If
		
	End Sub
	Private Sub cmdCancel_Click()
		On Error Resume Next
		
		mbEditFlag = False
		mbAddNewFlag = False
		adoPrimaryRS.CancelUpdate()
		If mvBookMark > 0 Then
			adoPrimaryRS.Bookmark = mvBookMark
		Else
			adoPrimaryRS.MoveFirst()
		End If
		mbDataChanged = False
		
	End Sub
	
	'UPGRADE_NOTE: update was upgraded to update_Renamed. Click for more: 'ms-help://MS.VSCC.v90/dv_commoner/local/redirect.htm?keyword="A9E4979A-37FA-4718-9994-97DD76ED70A7"'
	Private Sub update_Renamed()
		On Error GoTo UpdateErr
		
		adoPrimaryRS.UpdateBatch(ADODB.AffectEnum.adAffectAll)
		
		If mbAddNewFlag Then
			adoPrimaryRS.MoveLast() 'move to the new record
		End If
		
		mbEditFlag = False
		mbAddNewFlag = False
		mbDataChanged = False
		
		Exit Sub
UpdateErr: 
		MsgBox(Err.Description)
	End Sub
	
	Private Sub cmdClose_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdClose.Click
		update_Renamed()
		Me.Close()
	End Sub
	
	Private Sub goFirst()
		On Error GoTo GoFirstError
		
		adoPrimaryRS.MoveFirst()
		mbDataChanged = False
		
		Exit Sub
		
GoFirstError: 
		MsgBox(Err.Description)
	End Sub
	
	Private Sub goLast()
		On Error GoTo GoLastError
		
		adoPrimaryRS.MoveLast()
		mbDataChanged = False
		
		Exit Sub
		
GoLastError: 
		MsgBox(Err.Description)
	End Sub
	
    'Private Sub grdDataGrid_CellValueChanged(ByVal eventSender As System.Object, ByVal eventArgs As AxMSDataGridLib.DDataGridEvents_CellValueChangedEvent) Handles grdDataGrid.CellValueChanged
    '    If grdDataGrid.Columns(ColIndex).DataFormat.Format = "#,##0.00" Then
    '       grdDataGrid.Columns(ColIndex).DataFormat = 0
    '    End If

    'End Sub
End Class